modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectDefinitionLocalServiceTest.java

```diff
@@ -1738,80 +1818,108 @@ public void testUpdateSystemObjectDefinition() throws Exception {
+				ObjectDefinitionConstants.SCOPE_SITE));
+
 		// Modifiable system object definition must be published to be actived
 
-		_assertFailure(
+		AssertUtils.assertFailure(
 			ObjectDefinitionActiveException.class,
 			"Object definitions must be published before being activated",
-			objectDefinition,
-			objectDefinitionId ->
-				_objectDefinitionLocalService.updateCustomObjectDefinition(
-					null, objectDefinitionId, 0, 0, 0, 0, false, true, false,
-					true, false, false,
-					LocalizedMapUtil.getLocalizedMap("Charlie"), "Charlie",
-					null, null, false,
-					LocalizedMapUtil.getLocalizedMap("Charlies"),
-					ObjectDefinitionConstants.SCOPE_SITE));
+			() -> _objectDefinitionLocalService.updateCustomObjectDefinition(
+				null, objectDefinition1.getObjectDefinitionId(), 0, 0, 0, 0,
+				false, true, false, true, false, false,
+				LocalizedMapUtil.getLocalizedMap("Charlie"), "Charlie", null,
+				null, false, LocalizedMapUtil.getLocalizedMap("Charlies"),
+				ObjectDefinitionConstants.SCOPE_SITE));
 
 		// Label is null
 
-		_assertFailure(
+		AssertUtils.assertFailure(
 			ObjectDefinitionLabelException.class,
 			"Label is null for locale " + LocaleUtil.US.getDisplayName(),
-			objectDefinition,
-			objectDefinitionId ->
-				_objectDefinitionLocalService.updateCustomObjectDefinition(
-					null, objectDefinitionId, 0, 0, 0, 0, false, false, false,
-					true, false, false, null, "Charlie", null, null, false,
-					LocalizedMapUtil.getLocalizedMap("Charlie"),
-					ObjectDefinitionConstants.SCOPE_SITE));
+			() -> _objectDefinitionLocalService.updateCustomObjectDefinition(
+				null, objectDefinition1.getObjectDefinitionId(), 0, 0, 0, 0,
+				false, false, false, true, false, false, null, "Charlie", null,
+				null, false, LocalizedMapUtil.getLocalizedMap("Charlie"),
+				ObjectDefinitionConstants.SCOPE_SITE));
 
 		// Plural label is null
 
-		_assertFailure(
+		AssertUtils.assertFailure(
 			ObjectDefinitionPluralLabelException.class,
 			"Plural label is null for locale " + LocaleUtil.US.getDisplayName(),
-			objectDefinition,
-			objectDefinitionId ->
-				_objectDefinitionLocalService.updateCustomObjectDefinition(
-					null, objectDefinitionId, 0, 0, 0, 0, false, false, false,
-					true, false, false,
-					LocalizedMapUtil.getLocalizedMap("Charlie"), "Charlie",
-					null, null, false, null,
-					ObjectDefinitionConstants.SCOPE_SITE));
+			() -> _objectDefinitionLocalService.updateCustomObjectDefinition(
+				null, objectDefinition1.getObjectDefinitionId(), 0, 0, 0, 0,
+				false, false, false, true, false, false,
+				LocalizedMapUtil.getLocalizedMap("Charlie"), "Charlie", null,
+				null, false, null, ObjectDefinitionConstants.SCOPE_SITE));
+
+		_objectDefinitionLocalService.deleteObjectDefinition(objectDefinition1);
 
 		// After Update a modifiable system object definition check its
 		// properties
 
-		objectDefinition =
+		ObjectDefinition objectDefinition2 =
```

Comment:
Why did you need to create this new variable?

Reply:
Lambda require a final variable.